src/include/platform/ConnectivityManager.h

```diff
@@ -162,6 +163,9 @@ class ConnectivityManager
     void SetDelegate(ConnectivityManagerDelegate * delegate) { mDelegate = delegate; }
     ConnectivityManagerDelegate * GetDelegate() const { return mDelegate; }
 
+    void SetDriver(chip::DeviceLayer::NetworkCommissioning::WiFiDriver * driver) { mDriver = driver; }
```

Comment:
Why is mDriver on here?  It's never used by this class.

Reply:
Initially I added these new APIs into the NetworkCommissioning headers of all vendors because I wanted to have them as virtual methods to the WiFiDriver class, but instead I opted to a solution that I think is cleaner, adding to the GenericConnectivityManagerImpl to be easily shared by the platforms.
mDriver is required because I wanted to have access to methods like ConnectNetwork, AddOrUpdateNetwork, ScanNetworks and RemoveNetwork which are available through the NetworkCommissioning class.
The WiFi shell source file uses these methods and to make sure that it doesn't crash the application, I added checks for:
`
    if (ConnectivityMgr().GetDriver() == nullptr)
    {
        return CHIP_ERROR_NOT_IMPLEMENTED;
    }
`
Our application uses the call to SetDriver in order to set mDriver to the NetworkCommissioning instance used by our examples
`ConnectivityMgrImpl().SetDriver(&(::chip::DeviceLayer::NetworkCommissioning::NXPWiFiDriver::GetInstance()));`

So the main idea was to use the ConnectivityMgr instance of the ConnectivityManager class in WiFi.cpp to access the methods in a linked NetworkCommissioning instance, this way keeping it simple in the source file.
Do you have a suggestion on how this could be done in another way?

Reply:
The question is: should this be on the connectivity manager at all?

Reply:
The question is whether the connectivity manager should be talking to the driver at all.

Reply:
Right, most WiFiDrivers talk to `ConnectivityManager`, not the other way around, so this introduces a circular dependency. I understand there might not be an easy way to get the current Wi-Fi driver as it may be instantiated by an application, but your solution still assumes that you provide the driver to `ConnectivityManager` somehow, so wouldn't it be more straightforward to add an equivalent API to set the driver to be used by shell commands and not modify the `ConnectivityManager` interface at all?

I mean some setter like:
```
namespace chip::Shell {
   void SetWiFiDriver(WiFiDriver*);
}
```.